Calendar/Get-CalendarDiagnosticObjectsSummary.ps1

```diff
@@ -1234,67 +1255,155 @@ function BuildTimeline {
-    BuildCSV;
-    BuildTimeline;
+    # Get the CalLogs for each MeetingID found.
+    if ($GlobalObjectIds.count -gt 1) {
+        Write-Host "Found multiple GlobalObjectIds: $($GlobalObjectIds.Count)."
+        foreach ($MID in $GlobalObjectIds) {
+            Write-DashLineBoxColor  "Processing MeetingID: [$MID]"
+            $script:GCDO = GetCalendarDiagnosticObjects -Identity $Identity -MeetingID $MID;
+            Write-Verbose "Found $($GCDO.count) CalLogs with MeetingID[$MID] ."
+            BuildCSV -Identity $Identity;
+            BuildTimeline;
+        }
+    } else {
+        Write-Warning "No CalLogs were found.";
+    }
+}
+
+# ===================================================================================================
+# Main
+# ===================================================================================================
+
+$ValidatedIdentities = CheckIdentities -Identity $Identity
+
+if ($Subject -ne "") {
+    GetCalLogsWithSubject -Identity $ValidatedIdentities -Subject $Subject
+} elseif ($MeetingID -ne "") {
+
+    foreach ($ID in $ValidatedIdentities) {
+        #$script:GCDO = $script:InitialCDOs; # use the CalLogs that we already have, since there is only one.
+        #    $script:InitialCDOs = @(); # clear the Initial CDOs.
+        Write-DashLineBoxColor "Looking for CalLogs from [$ID] with passed in MeetingID."
+        Write-Verbose "Running: Get-CalendarDiagnosticObjects -Identity [$ID] -MeetingID [$MeetingID] -CustomPropertyNames $CustomPropertyNameList -WarningAction Ignore -MaxResults $LogLimit -ResultSize $LogLimit -ShouldBindToItem $true;"
+        $script:GCDO = GetCalendarDiagnosticObjects -Identity $ID -MeetingID $MeetingID;
+
+        if ($script:GCDO.count -gt 0) {
+            Write-Host "Found $($script:GCDO.count) CalLogs with MeetingID [$MeetingID]."
+            BuildCSV -Identity $ID;
+            BuildTimeline;
+        } else {
+            Write-Warning "No CalLogs were found for [$ID] with MeetingID [$MeetingID]."
+        }
+    }
 } else {
-    Write-Warning "A valid meeting ID was not found, manually confirm the meetingID";
+    Write-Warning "A valid MeetingID was not found, manually confirm the MeetingID";
 }
 
-Write-Host -ForegroundColor Yellow "`n`n`n============================================================================"
-Write-Host -ForegroundColor Yellow "Hope this script was helpful in getting (and understanding) the Calendar Logs."
-Write-Host -ForegroundColor Yellow "If you have issues or suggestion for this script,"
-Write-Host -ForegroundColor Yellow "`t please send them to <callogformatterdevs@microsoft.com>"
-Write-Host -ForegroundColor Yellow "============================================================================`n`n`n"
+Write-DashLineBoxColor  "Hope this script was helpful in getting and understanding the Calendar Logs.",
+"If you have issues or suggestion for this script, please send them to: ",
+"`t <Shanefe@Microsoft.com" -Color Yellow -DashChar =
```

Comment:
Recommend changing this to a group of some kind. Helps protects your email and in case you leave, the script doesn't need to be updated.

Reply:
Done.  I had to enable external addresses.